File: problems/0001-two-sum/main.rb
def two_sum(nums, target)
  hash = {}
  nums.each_with_index do |n, i|
    hash[n] = i
  end

  nums.each_with_index do |n, i|
    complement = target - n

    return [i, hash[complement]] if hash[complement] && hash[complement] != i
  end
end

File: problems/0001-two-sum/main.go
package main

import "fmt"

func twoSum(nums []int, target int) []int {
	hashTable := map[int]int{}
	for i, v := range nums {
		hashTable[v] = i
	}

	for i, v := range nums {
		complement := target - v
		res, ok := hashTable[complement]
		if ok && i != res {
			return []int{i, res}
		}
	}
	return []int{}
}

func main() {
	slices := []int{2, 7, 11, 15}
	fmt.Println(twoSum(slices, 9))
}
